LRAC-8122

91nilo opened the pull request:
https://issues.liferay.com/browse/LRAC-8122

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

91nilo commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-16.liferay.com/job/test-portal-source-format/5715/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-16.liferay.com//userContent/jobs/test-portal-source-format/builds/5715/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:sf - 0 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/61815a0f9c81d83f62c4d570f193385bef737ef3">61815a0f9c81d83f62c4d570f193385bef737ef3</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/91nilo/liferay-portal/tree/LRAC-8122">LRAC-8122</a><br/>Branch GIT ID: <a href="https://github.com/91nilo/liferay-portal/commit/fc6e931f562d41b00794040b2df68773c2e6e539">fc6e931f562d41b00794040b2df68773c2e6e539</a></p>0 out of 1jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-16.liferay.com/job/test-portal-source-format/5715/"><strike><strong>test-portal-source-format</strong></strike></a></li></ul></div><h5>For more details click <a href="https://test-1-16.liferay.com/userContent/jobs/test-portal-source-format/builds/5715/jenkins-report.html">here</a>.</h5><pre><code>     [java] java.lang.Exception: Found 2 formatting issues:
     [java] 1: ./portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DefinitionsEvents.testcase expected:&lt;...ibutesShownInAC \{
     [java] 		[var attributesList = '''
     [java] 			articleId,assetId,category,className,classPK,commentId,depth,
     [java] 			elementId,entryId,fieldName,fileEntryId,fileEntryUUID,focusDuration,
     [java] 			formId,groupId,href,numberOfWords,page,pageLoadTime,preview,
     [java] 			ratingType,score,sessionId,src,tagName,text,title,type,version,viewDuration
     [java] 		''';
     [java] 
     [java] 		var list = ListUtil.newListFromString("\$\{attributesList\}");
     [java] 
     [java] 		var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud();
     [java] 
     [java] 		ACUtils.launchAC();
     [java] 
     [java] 		ACNavigation.goToSettings();
     [java] 
     [java] 		ACSettings.goToDefinitions();
     [java] 
     [java] 		ACNavigation.goToDefinitionsEntity(entityName = "Event Attributes");
     [java] 
     [java] 		ACNavigation.goToDefinitionsAttributesTabs(key_tabName = "Attributes");
     [java] 
     [java] 		ACUtils.showEntries(dropdownOption = "30");
     [java] 
     [java] 		ACUtils.assertItemsName(itemList = "\$\{list\}");]
     [java] 	\}
     [java] 
     [java] 	@description =...&gt; but was:&lt;...ibutesShownInAC \{
     [java] 		[
     [java] 		var attributesList = '''
     [java] 			articleId,assetId,category,className,classPK,commentId,depth,
     [java] 			elementId,entryId,fieldName,fileEntryId,fileEntryUUID,focusDuration,
     [java] 			formId,groupId,href,numberOfWords,page,pageLoadTime,preview,
     [java] 			ratingType,score,sessionId,src,tagName,text,title,type,version,viewDuration
     [java] 		''';
     [java] 		
     [java] 		var list = ListUtil.newListFromString("\$\{attributesList\}");
     [java] 
     [java] 		var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud();
     [java] 
     [java] 		ACUtils.launchAC();
     [java] 
     [java] 		ACNavigation.goToSettings();
     [java] 
     [java] 		ACSettings.goToDefinitions();
     [java] 
     [java] 		ACNavigation.goToDefinitionsEntity(entityName = "Event Attributes");
     [java] 
     [java] 		ACNavigation.goToDefinitionsAttributesTabs(key_tabName = "Attributes");
     [java] 
     [java] 		ACUtils.showEntries(dropdownOption = "30");
     [java] 
     [java] 		ACUtils.assertItemsName(itemList = "\$\{list\}");
     [java] ]
     [java] 	\}
     [java] </code></pre></details></html>

91nilo commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-30.liferay.com/job/test-portal-source-format/41/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-30.liferay.com//userContent/jobs/test-portal-source-format/builds/41/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-16.liferay.com/job/test-portal-source-format/5715/">test-portal-source-format#5715</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-16.liferay.com/userContent/jobs/test-portal-source-format/builds/5715/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/alexcandidos/liferay-portal/pull/504">alexcandidos#504</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1551114981">[master] ci:test:sf - 91nilo &gt; alexcandidos - PR#504 - 2021-08-02[14:52:27]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-16.liferay.com/job/publish-testray-report/1599/">publish-testray-report#1599</a><br/></div></div>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 2 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/61815a0f9c81d83f62c4d570f193385bef737ef3">61815a0f9c81d83f62c4d570f193385bef737ef3</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/91nilo/liferay-portal/tree/LRAC-8122">LRAC-8122</a><br/>Branch GIT ID: <a href="https://github.com/91nilo/liferay-portal/commit/a8202f7320befbf4a0b20353563aa8a951b4aa1a">a8202f7320befbf4a0b20353563aa8a951b4aa1a</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-30.liferay.com/job/test-portal-source-format/41/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-30.liferay.com/userContent/jobs/test-portal-source-format/builds/41/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-30.liferay.com/job/test-portal-source-format/41/">test-portal-source-format#41</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-30.liferay.com/userContent/jobs/test-portal-source-format/builds/41/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/alexcandidos/liferay-portal/pull/504">alexcandidos#504</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=1551114990">[master] ci:test:sf - 91nilo &gt; alexcandidos - PR#504 - 2021-08-02[14:58:39]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-30.liferay.com/job/publish-testray-report/144/">publish-testray-report#144</a><br/></div></div>

joshitagaki commented on portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DefinitionsEvents.testcase:
```diff
@@ -35,12 +35,32 @@ definition {
+
+		ACSettings.goToDefinitions();
+
+		ACNavigation.goToDefinitionsEntity(entityName = "Event Attributes");
```
Hi @91nilo, there's a macro that should do the same thing here: https://github.com/alexcandidos/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACDefinitions.macro#L24

calvinctvo commented on portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACNavigation.macro:
```diff
@@ -32,6 +32,18 @@ definition {
 		ACNavigation.goToSidebarOption(sidebarOption = "assets");
 	}
 
+	macro goToDefinitionsAttributesTabs {
```
Hi @91nilo , I think these 2 macros should be in ACSettings.macro file since most interactions are in AC Settings.

calvinctvo commented on portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACUtils.macro:
```diff
@@ -23,6 +23,15 @@ definition {
 	}
 
+	@summary = "Asserts a list of items by the name"
+	macro assertItemsName {
```
Hi @91nilo , I think this should be change to something like assertAttributeList or assertItemsList. Since the macro is assert a list of items.

alexcandidos commented on portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACNavigation.path:
```diff
@@ -27,6 +27,26 @@
+</tr>
+<tr>
+	<td>ENTRIES_LIST_DROPDOWN_OPTION</td>
+	<td>//div[contains(@class, 'dropdown-menu show')]//a[text()='${key_dropdownOption}']</td>
```
Hi @91nilo , when using `contains`, please try to not use multiple options, in this case multiple classes, since it will let your code more fragile and susceptible to break for a change on the front-end.

So, I would suggest you to use something like:
`<td>//div[contains(@class, 'dropdown-menu') and contains (@class,'show')]//a[text()='${key_dropdownOption}']</td>`

or try to see if you are able to create a path with a single class element (the best way).

So, if the order of the classes changes, your path will not break.

91nilo commented:
@calvinctvo @alexcandidos @joshitagaki 
Hey guys, i did the following changes:

- Used the macro goToDefinitionType instead of creating this new one;
- Removed the goToDefinitionsEntity macro and path
-  Moved goToDefinitionsAttributesTabs from ACNavigation to ACSettings;
- Changed assertItemsName macro name to assertItemsList
- Changed the tests using this macro to the new one (assertItemsList) and ran these tests;
-  Changed the ENTRIES_LIST_DROPDOWN_OPTION to make it less susceptible to break.

Thanks for the feedbacks, if there's anything else i could do to improve the code, just let me know!

91nilo commented:
Sending this PR on https://github.com/alexcandidos/liferay-portal/pull/505
